prime-router/src/main/kotlin/Report.kt

```diff
@@ -721,6 +723,7 @@ class Report {
             sendingFacility: String = "cdcprime",
             processingModeCode: String = "T",
             translationConfig: TranslatorConfiguration? = null,
+            fileName: FileName? = null
```

Comment:
I suggest you change all of these to nameFormatTemplates configs outside the code.   Add more functions (eg uuid()) as needed.   Even the standard format should be specified as a template outside the code.

If you really still want to have named formats, then have a yml file that declares a bunch of predefined named formats.   That way, let's say I invent a format for Utah.    Then later, I discover everybody wants to use Utah's format.   Then I just move it to my `predefinedNameFormats.yml` file, and its available globally.

Reply:
I like this idea a lot, and I think it's the right direction to head with onboarding. They just become another type of metadata with a name.